### PR TITLE
Track recent files and allow opening them

### LINKED ISSUE #3855: Add Show Recent Files dialog
Suggested by King employee at induction in Stockholm, April 2019.

like IDEA ctrl/command+E

!Screenshot 2019-04-12 at 07.31.41.png!

### PR DESCRIPTION
**Pull request by vlaaad**
User-facing changes:

User now can press `Cmd+Shift+T` (macOS) / `Ctrl+Shift+T` (Windows and Linux) to reopen previously closed tabs. File menu now has a new submenu — Recent Files — that allows selecting a recently accessed file to open/focus on. Also, editor now has Recent Files dialog that allows searching from a larger list of recent files, available using `Ctrl+R` (macOS) / `Alt+R` (Windows and Linux) shortcuts.

Fixes #3855

### CODE REVIEW

File: editor/src/clj/editor/ui.clj [hunk 1]
```diff
@@ -1324,7 +1324,7 @@
                                 label
                                 icon
                                 style-classes
-                                (make-menu-items scene options command-contexts {} evaluation-context)
+                                (make-menu-items scene options command-contexts command->shortcut evaluation-context)
```
Comment (vlaaad) [inline editor/src/clj/editor/ui.clj:1327]: Not sure why we didn't use `command->shortcut` in the submenus, but without that we didn't show shortcuts in nested submenus.

**Review by matgis — APPROVED** [review]
Works great! 👍

I did notice the icons in the resource picker dialog looking blurry compared to the ones in the Assets panel, but that issue seems to have been there for some time.

**Comment by matgis** [comment]
Also, is there an easy way we can highlight the matching substrings in the filter dialog like we do in the other asset dialogs? Maybe it could be a separate PR?

image

File: editor/src/clj/editor/recent_files.clj [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+       :ok-label "Open"
+       :cell-fn (fn [[resource view-type]]
+                  {:style-class (into ["list-cell"] (resource/style-classes resource))
+                   :graphic {:fx/type fx.h-box/lifecycle
+                             :style-class "content"
+                             :alignment :center-left
+                             :spacing 4
+                             :children [{:fx/type ui/image-icon
+                                         :path (workspace/resource-icon resource)
+                                         :size 16}
```
[the remark this answers is not in the document]
Comment (vlaaad) [reply editor/src/clj/editor/recent_files.clj]: Hmm, there was a discrepancy in behavior between `editor.ui/image-icon` and `editor.icons/get-image-view` (the latter always loaded icons in full size), I fixed it.

**Comment by vlaaad** [comment]
@matgis I added highlighting in this PR :)

**Pull request merged**